datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -753,8 +753,14 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
             BinaryExpr {
                 left,
                 op: Divide,
+                right: _,
+            } if is_null(&left) => *left,
+            // null / A --> null
+            BinaryExpr {
+                left: _,
+                op: Divide,
                 right,
-            } if left == right && is_null(&left) => *left,
+            } if is_null(&right) => *right,
```

Comment:
Isn't this checking for `A / null` (not `null / A` as the comments suggest)?
 (as in I think the comments are backwards)

Reply:
Thank you for catching this, I will update.

Reply:
Done.